# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/HttpChannelChild.h
# Commit: 3b150d676b31
# Full Hash: 3b150d676b3122bffa703813876d1fbc35e10d17
# Author: Randell Jesup <rjesup@mozilla.com>
# Date: 2024-03-19 16:41:28
# Regressor Bug: 1356686
# File Overlap Count: 1
# Description:
#   Bug 1356686 - Do decompression off main thread in content r=necko-reviewers,valentin,extension-reviewers,robwu
#   
#   Content processes will now always retarget delivery of OnDataAvailable for Http
#   channels off the main thread. Consumers that were previously redirecting
#   off-main thread are not affected and their retargeting will stick, but any
# ==============================================================================

diff -r 54754a50d992 -r 3b150d676b31 netwerk/protocol/http/HttpChannelChild.h
--- a/netwerk/protocol/http/HttpChannelChild.h	Mon Mar 18 20:04:00 2024 +0000
+++ b/netwerk/protocol/http/HttpChannelChild.h	Mon Mar 18 20:04:01 2024 +0000
@@ -33,7 +33,7 @@
 #include "nsIThreadRetargetableRequest.h"
 #include "mozilla/net/DNS.h"
 
-using mozilla::Telemetry::LABELS_HTTP_CHILD_OMT_STATS;
+using mozilla::Telemetry::LABELS_HTTP_CHILD_OMT_STATS_2;
 
 class nsIEventTarget;
 class nsIInterceptedBodyCallback;
@@ -271,6 +271,9 @@
 
   nsresult MaybeLogCOEPError(nsresult aStatus);
 
+  void RetargetDeliveryToImpl(nsISerialEventTarget* aNewTarget,
+                              MutexAutoLock& aLockRef);
+
  private:
   // this section is for main-thread-only object
   // all the references need to be proxy released on main thread.
@@ -312,8 +315,8 @@
 
   // The result of RetargetDeliveryTo for this channel.
   // |notRequested| represents OMT is not requested by the channel owner.
-  Atomic<LABELS_HTTP_CHILD_OMT_STATS, mozilla::Relaxed> mOMTResult{
-      LABELS_HTTP_CHILD_OMT_STATS::notRequested};
+  Atomic<LABELS_HTTP_CHILD_OMT_STATS_2, mozilla::Relaxed> mOMTResult{
+      LABELS_HTTP_CHILD_OMT_STATS_2::notRequested};
 
   uint32_t mCacheKey{0};
   int32_t mCacheFetchCount{0};